internal_ws/ykcompile/src/lib.rs

```diff
@@ -1093,7 +1100,7 @@ impl TraceCompiler {
     fn c_cast(&mut self, dest: &IPlace, src: &IPlace) {
         let src_loc = self.iplace_to_location(src);
         let ty = &*SIR.ty(&src.ty()); // Type of the source.
-        let cty = SIR.ty(&dest.ty()); // Type of the cast (same as dest type).
+        let cty = SIR.ty(&dest.ty()); // Type of the cast (same as dest type XXX isn't this the case by definition?).
```

Comment:
Yes. I think I just wanted to make sure it's clear why we didn't use the explicit type from the cast statement, which we remove during SIR construction for this very reason. But I agree that we can just remove the stuff in the brackets.

Reply:
Fixed in db5a334.